gax/src/main/java/com/google/api/gax/batching/v2/BatcherImpl.java

```diff
@@ -159,6 +170,57 @@ public void close() throws InterruptedException {
     flush();
   }
 
+  private boolean isAnyThresholdReached(ElementT element) {
+    for (BatchingThreshold<ElementT> threshold : thresholds) {
+      threshold.accumulate(element);
+      if (threshold.isThresholdReached()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private void resetThresholds() {
+    for (int i = 0; i < thresholds.size(); i++) {
+      thresholds.set(i, thresholds.get(i).copyWithZeroedValue());
+    }
+  }
+
+  private ImmutableList<BatchingThreshold<ElementT>> getThresholds(
```

Comment:
To be honest the whole element counter/threshold abstraction feels a bit overkill. Maybe it can be simplified to the Batch simply keeping track of the number of elements it has and the bytesize? Then you can add a simple method isFull()?

Reply:
Thanks for the feedback!! 
I have refactored BatcherImpl to include `hasAnyThresholdReached()`. I believe this should simplify the threshold check.